Lib/test/test_zipfile/test_core.py

```diff
@@ -1801,17 +1801,18 @@ def test_write_with_source_date_epoch(self):
                     self.assertAlmostEqual(z_time, g_time, delta=1)
 
     def test_write_without_source_date_epoch(self):
-        if 'SOURCE_DATE_EPOCH' in os.environ:
-            del os.environ['SOURCE_DATE_EPOCH']
+        with os_helper.EnvironmentVarGuard() as env:
+            if 'SOURCE_DATE_EPOCH' in env:
```

Comment:
nit: Could this be `del env['SOURCE_DATE_EPOCH']` (`__delitem__` in EnvironmentVarGuard seems to handle both exists in original environment and not)